vertx-web/src/main/java/io/vertx/ext/web/impl/RouteImpl.java

```diff
@@ -232,6 +232,7 @@ synchronized boolean matches(RoutingContext context, String mountPoint, boolean
     }
     HttpServerRequest request = context.request();
     if (!methods.isEmpty() && !methods.contains(request.method())) {
+      ((RoutingContextImplBase) context).setDefaultErrorCode(405);
```

Comment:
you will get the error by parsing order:

1. method
2. path
3. content-type

I think this makes sense and keeps things simple (from our side). In this case if no other handler matches the first failure will be the method that is not expected only `GET` is accepted.